api/post.go

```diff
@@ -728,6 +730,92 @@ func updateMentionCountAndForget(channelId, userId string) {
 	}()
 }
 
+func checkForOutOfChannelMentions(c *Context, post *model.Post, channel *model.Channel) {
```

Comment:
That actually sounds a lot more useful than doing it after the user has actually made the post. I was just following the ticket which is how Slack does it, but I'll put in a feature request for just having a little popup when you actually type in the @mention since that would actually remove the need to have the system message.